staging/src/k8s.io/client-go/util/certificate/certificate_store.go

```diff
@@ -67,14 +68,30 @@ type FileStore interface {
 // updates will be written to the ${certDirectory} directory and
 // ${certDirectory}/${pairNamePrefix}-current.pem will be created as a soft
 // link to the currently selected cert/key pair.
+//
+// Contextual logging: NewFileStoreWithLogger should be used instead of NewFileStore in code which supports contextual logging.
```

Comment:
Is this one not getting deprecated?

Reply:
No, not all users are required to support contextual logging. For those who don't care, continuing to use `NewFileStore` is fine. We don't want to create unnecessary churn in the ecosystem.